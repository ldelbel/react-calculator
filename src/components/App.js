import React, { useState } from 'react';
import '../App.css';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/Calculate';

const App = () => {
  const [state, setState] = useState({ total: null, next: null, operation: null });
  const { total, next, operation } = state;

  const handleClick = buttonName => {
    setState(calculate({ total, next, operation }, buttonName));
  };

  return (
    <>
      <Display result={next || total || '0'} />
      <ButtonPanel clickHandler={handleClick} />
    </>
  );
};

export default App;
