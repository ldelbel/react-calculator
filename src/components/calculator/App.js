import React, { useState } from 'react';
import '../../App.css';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../../logic/calculate';

const App = () => {
  const [state, setState] = useState({ total: null, next: null, operation: null });
  const { total, next, operation } = state;

  const handleClick = buttonName => {
    setState(calculate({ total, next, operation }, buttonName));
  };

  return (
    <div style={{ width: '700px' }}>
      <Display result={next || total || '0'} />
      <ButtonPanel clickHandler={handleClick} />
    </div>
  );
};

export default App;
