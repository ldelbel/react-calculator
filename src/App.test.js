import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './components/calculator/App';

test('renders learn react link', () => {
  render(<App />);
  const app = renderer.create(<App />).toJSON();
  expect(app).toMatchSnapshot();
});
