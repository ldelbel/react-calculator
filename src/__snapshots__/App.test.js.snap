// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  style={
    Object {
      "width": "700px",
    }
  }
>
  <h1
    style={
      Object {
        "alignItems": "center",
        "background": "#858694",
        "color": "white",
        "display": "flex",
        "fontSize": "2.7em",
        "height": "100px",
        "justifyContent": "flex-end",
        "margin": "0",
        "padding": "0 20px",
      }
    }
  >
    0
  </h1>
  <div
    style={
      Object {
        "height": "100px",
      }
    }
  >
    <button
      className="sc-fFubgz fFlHsM"
      onClick={[Function]}
      type="submit"
    >
      AC
    </button>
    <button
      className="sc-bkzZxe emezma"
      onClick={[Function]}
      type="submit"
    >
      +/-
    </button>
    <button
      className="sc-idOhPF hUcZIW"
      onClick={[Function]}
      type="submit"
    >
      %
    </button>
    <button
      className="sc-dIUggk jwtOzR"
      onClick={[Function]}
      type="submit"
    >
      รท
    </button>
  </div>
  <div
    style={
      Object {
        "height": "100px",
      }
    }
  >
    <button
      className="sc-hHftDr igjhEO"
      onClick={[Function]}
      type="submit"
    >
      7
    </button>
    <button
      className="sc-dmlrTW ckXJci"
      onClick={[Function]}
      type="submit"
    >
      8
    </button>
    <button
      className="sc-kfzAmx hBRgxl"
      onClick={[Function]}
      type="submit"
    >
      9
    </button>
    <button
      className="sc-fKFyDc djQRsb"
      onClick={[Function]}
      type="submit"
    >
      x
    </button>
  </div>
  <div
    style={
      Object {
        "height": "100px",
      }
    }
  >
    <button
      className="sc-bBXqnf fenkYT"
      onClick={[Function]}
      type="submit"
    >
      4
    </button>
    <button
      className="sc-iwyYcG dtpknk"
      onClick={[Function]}
      type="submit"
    >
      5
    </button>
    <button
      className="sc-cxFLnm hqaqBk"
      onClick={[Function]}
      type="submit"
    >
      6
    </button>
    <button
      className="sc-lmoMRL cyteKH"
      onClick={[Function]}
      type="submit"
    >
      -
    </button>
  </div>
  <div
    style={
      Object {
        "height": "100px",
      }
    }
  >
    <button
      className="sc-iJuUWI gRxHnv"
      onClick={[Function]}
      type="submit"
    >
      1
    </button>
    <button
      className="sc-giIncl jhGwHE"
      onClick={[Function]}
      type="submit"
    >
      2
    </button>
    <button
      className="sc-ezrdKe kkGuSJ"
      onClick={[Function]}
      type="submit"
    >
      3
    </button>
    <button
      className="sc-bYEvPH jtFciC"
      onClick={[Function]}
      type="submit"
    >
      +
    </button>
  </div>
  <div
    style={
      Object {
        "height": "100px",
      }
    }
  >
    <button
      className="sc-kLgntA bxPtXP"
      onClick={[Function]}
      type="submit"
    >
      0
    </button>
    <button
      className="sc-iktFzd iGORtU"
      onClick={[Function]}
      type="submit"
    >
      .
    </button>
    <button
      className="sc-jJEJSO kUKBip"
      onClick={[Function]}
      type="submit"
    >
      =
    </button>
  </div>
</div>
`;
